pkg/tlsutils/tlsconfig.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tlsutils
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// CreateTLSConfig creates tls.Config instance from TLS parameters passed in environment variables with the given prefix
+func CreateTLSConfig(prefix string) (*tls.Config, error) {
+	caFile := os.Getenv(fmt.Sprintf("%s_CA_FILE", prefix))
+	certFile := os.Getenv(fmt.Sprintf("%s_CERT_FILE", prefix))
+	keyFile := os.Getenv(fmt.Sprintf("%s_KEY_FILE", prefix))
+	serverName := os.Getenv(fmt.Sprintf("%s_TLS_SERVER_NAME", prefix))
+	isInsecureStr := strings.ToLower(os.Getenv(fmt.Sprintf("%s_TLS_INSECURE", prefix)))
```

Comment:
> Those variables are part of provider configuration. --help doesn't show you all the environment variables that can be used to control AWS/GCE etc providers, so why should it do it for designate?

I don't see any special environment variables in any of those provides. I only see something in the Cloudflare provider, which I also believe should be moved to the main config.

> For designate there are about 15 environment variables that affect how and where it connects and authenticates. For some of other providers it is not better. I don't think that extracting parameters of all providers and merging them into one flat list and then printing in --help screen is a good idea.

I think it's better than having them appear magically in the middle of a provider without any information about it in `--help`

> I'd rather think about extending provider interface so that providers can print their own help information. Then I'd be able to say something like external-dns --help --provider designate and get designate specific help with all those variables listed.

This could be a good solution, maybe a bit tricky with the current CLI library.

In the end this is a great PR and I'm just pointing out something I believe could be done in a better way. If you don't agree it's fine :). I'll let @linki and @ideahitme review it.